mr_proper/main.py

```diff
@@ -7,29 +8,42 @@
 
 def parse_args() -> argparse.Namespace:
     parser = argparse.ArgumentParser()
-    parser.add_argument('filename', type=str)
+    parser.add_argument('objectname', type=str)
     parser.add_argument('--recursive', action='store_true')
     return parser.parse_args()
 
 
-def main() -> None:
-    args = parse_args()
-    ast_tree = get_ast_tree(args.filename)
+def checking_function(args: argparse.Namespace) -> None:
+    ast_tree = get_ast_tree(args.objectname)
     if not ast_tree:
         sys.stdout.write('Error parsing ast tree\n')
         return
+    sys.stdout.write(f'{args.objectname}:\n')
     for funcdef_node in get_all_funcdefs_from(ast_tree):
         function_name = funcdef_node.name
         is_pure, pureness_errors = is_function_pure(
             funcdef_node,
             ast_tree,
             with_errors=True,
             recursive=args.recursive,
-            pyfilepath=args.filename,
+            pyfilepath=args.objectname,
         )
         if is_pure:
             sys.stdout.write(f'{function_name} is pure!\n')
         else:
             sys.stdout.write(f'{function_name} is not pure because of:\n')
             for error in pureness_errors:
                 sys.stdout.write(f'\t{error}\n')
+
+
+def main() -> None:
+    args = parse_args()
+    path_to_files = [args.objectname]
+    if os.path.isdir(args.objectname):
```

Comment:
What happens if path is dir, but it doesn't exists?
This is very common case, so it would be nice to make it explicit.

Reply:
It seems to be done!:)